pkg/storagev2/migration/index.go

```diff
@@ -0,0 +1,128 @@
+// Copyright 2022 The Swarm Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package migration
+
+import (
+	storage "github.com/ethersphere/bee/pkg/storagev2"
+)
+
+type (
+	// ItemDeleteFn is callback function called in migration step
+	// to check if Item should be removed in this step.
+	ItemDeleteFn func(storage.Item) (delete bool)
+
+	// ItemUpdateFn is callback function called in migration step
+	// to check if Item should be updated in this step.
+	ItemUpdateFn func(storage.Item) (updateditem storage.Item, hasChanged bool)
+)
+
+// WithItemDeleteFn return option with ItemDeleteFn set.
+func WithItemDeleteFn(fn ItemDeleteFn) option {
+	return func(o *options) {
+		o.deleteFn = fn
+	}
+}
+
+// WithItemUpdaterFn return option with ItemUpdateFn set.
+func WithItemUpdaterFn(fn ItemUpdateFn) option {
+	return func(o *options) {
+		o.updateFn = fn
+	}
+}
+
+type option func(*options)
+
+type options struct {
+	deleteFn ItemDeleteFn
+	updateFn ItemUpdateFn
+}
+
+func newOptions() *options {
```

Comment:
The name is a little bit misleading, it should be something like `defaultOptions` or `noopOptions`. Also, the procedure that this function wraps is not complex or used in other places, so it can be nested. The same goes for the `applyAll` function.

Reply:
More often then not, types created with `func NewType() Type` or just `func New() Type` methods have some default values set (depending on the use case). That is the reason for having these functions (to create type with default values), because creating zero value types is usually done via semantics `Type{}` or `new(Type)`. Some devs like having `New` funcs as constructor for zero values as well, so in that case `New` func creates type with zero value or default values, whichever suits more usage of particular type.

However, I have updated name to `defaultOptions` as suggested.

----

Could you explain other suggestion?

Reply:
The other suggestions have become irrelevant because this [CL](https://github.com/ethersphere/bee/pull/3421/commits/cbd92d3cc30b36b09bc0a358e492272708e19613) made after this comment contains new functions that use those functions.